code/datums/components/directional_attack.dm

```diff
@@ -0,0 +1,154 @@
+/datum/component/directional_attack
+	var/active = TRUE
+	var/datum/action/directional_attack_toggle/toggle_action
+
+/datum/component/directional_attack/Initialize()
+	. = ..()
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	toggle_action = new()
+	var/toggle_path
+	toggle_path = .proc/living_activation_toggle
+	toggle_action.give_action(parent)
+	toggle_action.update_button_icon(active)
+	RegisterSignal(toggle_action, COMSIG_ACTION_TRIGGER, toggle_path)
+	if(active)
+		RegisterSignal(parent, COMSIG_MOB_CLICKON, .proc/select_directional_action)
+
+/datum/component/directional_attack/Destroy(force, silent)
+    QDEL_NULL(toggle_action)
+    return ..()
+
+/datum/component/directional_attack/proc/select_directional_action(datum/source, atom/A, params)
+	if(ishuman(parent))
+		human_directional_action(A, params)
+	else if(isxeno(parent))
+		xeno_directional_action(A, params)
+	else
+		living_directional_action(A, params)
+
+/datum/component/directional_attack/proc/living_activation_toggle(datum/source)
+	var/mob/living/attacker = parent
+	active = !active
+	to_chat(attacker, "<span class='notice'>You will now [active ? "attack" : "not attack"] enemies in melee range upon clicking in their direction.</span>")
+	if(active)
+		RegisterSignal(attacker, COMSIG_MOB_CLICKON, .proc/select_directional_action)
+	else
+		UnregisterSignal(attacker, COMSIG_MOB_CLICKON, .proc/select_directional_action)
+	toggle_action.update_button_icon(active)
+
+/datum/component/directional_attack/proc/living_directional_action_checks(mob/living/L)
+	if(COOLDOWN_CHECK(src, COOLDOWN_DIRECTIONAL_ATTACK))
+		return NONE
+	if(L.throwing)
+		return NONE
+
+/datum/component/directional_attack/proc/carbon_directional_action_checks(mob/living/L)
+	var/mob/living/carbon/attacker = parent
+	. = living_directional_action_checks(L)
+	if(!isnull(.))
+		return
+	if(QDELETED(L))
+		return
+	switch(attacker.a_intent)
+		if(INTENT_HELP, INTENT_GRAB)
+			return NONE
+	if(attacker.get_active_held_item())
+		return NONE //We have something in our hand.
```

Comment:
Move this to living check since living has hands as well and you wont need to duplicate later.
```suggestion
```